components/script/dom/htmlelement.rs

```diff
@@ -897,6 +937,82 @@ impl HTMLElement {
             None => false,
         }
     }
+
+    /// <https://html.spec.whatwg.org/multipage/#rendered-text-fragment>
+    fn rendered_text_fragment(&self, input: DOMString) -> DomRoot<DocumentFragment> {
+        // Step 1: Let fragment be a new DocumentFragment whose node document is document.
+        let document = document_from_node(self);
+        let fragment = DocumentFragment::new(&document);
+
+        // Step 2: Let position be a position variable for input, initially pointing at the start
+        // of input.
+        let mut position = input.chars().peekable();
+
+        // Step : Let text be the empty string.
+        let mut text = String::new();
+
+        // Step 4
+        while let Some(ch) = position.next() {
+            match ch {
+                // While position is not past the end of input, and the code point at position is
```

Comment:
We have some code at https://github.com/servo/servo/blob/ebed9218f2907767ba3c9dd9f27f30a6a6e9f225/components/net/fetch/headers.rs#L88

It might make sense to move it to `components/shared/net` and re-use it here.

Reply:
I fear then we'd have to iterate over the string twice since we also need to turn linefeeds into `<br>` elements and generate multiple Text nodes